docs(block-node): update operator docs for per-service ports. (#3257)
hiero-ledger/hiero-block-node: docs/block-node/operations/network-ports-and-protocols.md

#### Notes

- A Block Node with no `BACKFILL_BLOCK_NODE_SOURCES_PATH` file mounted makes no outbound gRPC connections of this kind. The plugin, if present, loads but stays idle.
- The list of peer Block Nodes is operator-supplied via a JSON file mounted into the pod. Firewalls and security groups must permit egress to every listed peer's Subscribe API port (`40980` in LFH; `40840` in base-chart default).

### Health and readiness probes

> Kubernetes probes query the Block Node's dedicated health HTTP server with HTTP/1.1 GET requests. In the LFH profile this runs on its own port; the base chart defaults to the same shared port as the gRPC services.

Reviewer: The health port never runs on a shared port anymore.

```suggestion
Kubernetes probes query the Block Node's dedicated health HTTP server with HTTP/1.1 GET requests. This service always runs on a separate dedicated port.
```

|     Field      |                                                             Value                                                              |
|----------------|--------------------------------------------------------------------------------------------------------------------------------|

> | Port           | `40983` (`blockNode.ports.health`; set to `40983` in both base `values.yaml` and LFH profile — health always has its own port) |

Reviewer: We probably don't need this explanation here; it's clear enough above.

```suggestion
| Port           | `40983`                                               |
```

| Helm value     | `blockNode.ports.health`                                                                                                       |
| Env var        | `HEALTH_PORT`                                                                                                                  |
| Liveness path  | `/healthz/livez` (default; `blockNode.health.liveness.endpoint`)                                                               |
| Readiness path | `/healthz/readyz` (default; `blockNode.health.readiness.endpoint`)                                                             |
| Protocol       | HTTP/1.1 GET                                                                                                                   |
| Direction      | Inbound                                                                                                                        |
| Initiator      | Kubelet                                                                                                                        |

#### Notes

- Probe traffic is intra-cluster only - kubelet to pod IP. Cluster-external firewalls do not need a rule for it.

> - A `NetworkPolicy` that restricts ingress to specific ports must explicitly allow the kubelet to reach the health port (`40983` in LFH), or the probes will fail and Kubernetes will restart the pod.

Reviewer: ```suggestion
- A `NetworkPolicy` that restricts ingress to specific ports must explicitly allow the kubelet to reach the health port (`40983` default), or the probes will fail and Kubernetes will restart the pod.
```

### Tier 2 Block Node

Receives the block stream **from another Block Node** - typically a Tier 1, but a Tier 2 may also pull from another Tier 2. The publish path is replaced by a subscribe-from-upstream path; otherwise the network surface matches Tier 1.

|                 Flow                 |  Direction   |           Port            |                                                                                                                                       Notes                                                                                                                                       |
|--------------------------------------|--------------|---------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Pull from upstream Block Node        | **Outbound** | `40980` to peer (typical) | Mechanism is operator-dependent: per [Block Node Types](../../Block-Node-Types.md), Tier 2 may receive its stream via direct gRPC subscribe, gossip, file transfer, or another mechanism. A typical deployment uses the Subscribe API port (`40980` in LFH) against the upstream. |

Reviewer: Another explanation that probably belongs outside the table.

```suggestion
| Pull from upstream Block Node        | **Outbound** | `40980` to peer (typical) | Mechanism is operator-dependent. |
```

| Subscribe (Mirror / downstream → BN) | Inbound      | `40980` gRPC              | Same shape as Tier 1                                                                                                                                                                                                                                                              |
| Status (clients → BN)                | Inbound      | `40982` gRPC              | Subscribers and other clients query the Status API for available blocks, available services, and response latency                                                                                                                                                                 |
| Block Access (clients → BN)          | Inbound      | `40981` gRPC/HTTP         | Block retrieval API                                                                                                                                                                                                                                                               |

> | Metrics scrape                       | Inbound      | `16007` HTTP              | Intra-cluster                                                                                                                                                                                                                                                                     |

Reviewer: I wonder if we should have a note somewhere that this port is _never_ on the `40840` port because metrics are completely independent of the application level APIs and webservers.

The Block Node Helm chart does not ship a `NetworkPolicy` template or any other firewall manifest. Operators express the policy in whichever primitive their environment uses - Kubernetes `NetworkPolicy`, a cloud-provider security group, a host-level firewall, or a service mesh. The list below states what any such policy must allow or deny for a Block Node deployment, derived row-by-row from the [Port summary](#port-summary).

### Must allow (LFH production profile)

- **Inbound TCP `40984`** from Consensus Node source IPs only — restrict this port to known publisher IPs; deny all other inbound.
- **Inbound TCP `40980`** from Mirror Nodes and peer Block Nodes that subscribe to this Block Node.
- **Inbound TCP `40981`** from authorized block-access clients.
- **Inbound TCP `40982`** from monitoring and operator tooling (server-status API).
- **Inbound TCP `40983`** from the Kubernetes kubelet (liveness and readiness probes: `/healthz/livez`, `/healthz/readyz`). Probe traffic is intra-cluster only.
- **Inbound TCP `16007`** from the monitoring system that scrapes Prometheus metrics. Typically intra-cluster; rarely needs external exposure.

> - **Outbound TCP `40980`** to each peer Block Node listed in `BACKFILL_BLOCK_NODE_SOURCES_PATH`, if the `backfill` plugin has been configured with a non-empty sources file. A Block Node without a backfill sources file makes no such outbound connections.

Reviewer: A couple items:
1. There are more outbound connections for Mirror Node, for RSA bootstrap, for TSS Bootstrap.
1. RFH has connections to S3 storage as well.
1. Solo Provisioner will manage both inbound and outbound firewall rules in the next Solo Provisioner release when traffic shaping support is enabled.

> > **Base-chart default (non-LFH deployments).** When per-service ports are not set, replace the per-port rules above with a single **Inbound TCP `40840`** rule covering all services, and **Outbound TCP `40840`** for backfill egress. The kubelet health probe still uses **Inbound TCP `40983`** in this configuration — `health` is the one port that is set explicitly in the base `values.yaml` regardless of whether the LFH profile is active.

Reviewer: Here is a good place to note that metrics is also different in all cases.
